cyipopt/tests/unit/test_scipy_optional.py

```diff
@@ -131,6 +135,94 @@ def test_minimize_ipopt_jac_hessians_constraints_with_arg_kwargs():
     np.testing.assert_allclose(res.get("x"), expected_res, rtol=1e-5)
 
 
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
+@pytest.mark.parametrize('method', MINIMIZE_METHODS)
+def test_minimize_ipopt_jac_with_scipy_methods(method):
```

Comment:
This test is more complex than `test_minimize_ipopt_jac_hessians_constraints_with_arg_kwargs` because it is intended to be stronger. For instance, `test_minimize_ipopt_jac_hessians_constraints_with_arg_kwargs` passes even if `*a*b` is omitted from the functions, so it does not test whether the `a` and `b` are actually being used. It is also possible that the functions are never executed. The tests below would fail unless `args` is passed as a positional argument, `kwargs` is passed as keyword arguments, if the callable didn't run, and if the callable didn't affect the results.